src/tflite_cpu.cpp

```diff
@@ -494,10 +481,17 @@ class MLModelServiceTFLite : public vsdk::MLModelService,
     // All of the meaningful internal state of the service is held in
     // a separate state object to help ensure clean replacement of our
     // internals during reconfiguration.
-    struct state_ {
+    struct state_ final : public tflite::ErrorReporter {
         explicit state_(vsdk::Dependencies dependencies, vsdk::ResourceConfig configuration)
             : dependencies(std::move(dependencies)), configuration(std::move(configuration)) {}
 
+        int Report(const char *format, va_list args) override {
+            char buffer[4096];
+            static_cast<void>(vsnprintf(buffer, sizeof(buffer), format, args));
+            interpreter_error_data = buffer;
+            return 0;
```

Comment:
My read of [the docs](https://ai.google.dev/edge/api/tflite/cc/class/tflite/error-reporter) is that this function is supposed to return the number of characters written, which I imagine is `std::strlen(buffer)`. Admittedly, returning "number of characters written or zero on success, and negative number on error" is ambiguous: when do you return 0 and when do you return the number of characters written? The docs' comparison to `printf` reinforces my preference to return the number of characters written.

Reply:
The examples above show one case of returning zero, and another of returning number of bytes written. I expect the allowance for returning the number of characters written is to make it easier to just immediately return the result of delegating to a `printf` function. But the docs pretty clearly say it is ok to return zero.

Further evidence for this is that tflite defines an internal macro for reporting errors, which ignores the return value:
```
#define TF_LITE_REPORT_ERROR(reporter, ...)                               \
  do {                                                                    \
    static_cast<::tflite::ErrorReporter*>(reporter)->Report(__VA_ARGS__); \
  } while (false)
#else  // TF_LITE_STRIP_ERROR_STRINGS
```